waku/common/sqlite.nim

```diff
@@ -1,334 +1,7 @@
-{.push raises: [Defect].}
+{.push raises: [].}
```

Comment:
Thanks for starting to add these. Can you see any (good) reason for us to rather use:

```
when (NimMajor, NimMinor) < (1, 4):
  {.push raises: [Defect].}
else:
  {.push raises: [].}
```
?

Perhaps if someone compiles nwaku into a system with an older nim version (unlikely and should probably not be allowed) or (perhaps more likely) that the nimbus-build-system gets downgraded due to some unforeseen issues?

Reply:
I removed the `Defect` because the compiler added extra noise because of that pragma. I did it as an experiment, and I forgot to restore it.

Let's do this in a unique PR where we change all the occurrences. WDYT? 🤔